src/utils/formik-helpers.ts

```diff
@@ -30,18 +37,37 @@ export const hasTouchedError = (
   return fieldNames.some((fieldName) => {
     const fieldHasError = !!getIn(errors, fieldName);
     const fieldIsTouched = getIn(touched, fieldName);
-    return fieldHasError && fieldIsTouched;
+    // Touched data can be inconsistent with array of null or undefined elements when adding elements dynamically
+    // to a FieldArray, so check for value to be true explicitly, otherwise an empty array will also be true
+    return fieldHasError && fieldIsTouched === true;
   });
 };
 
-export const getAllFileFields = (numberOfUploadedFiles: number) => {
+export const getAllFileFields = (fileSet: File[]) => {
   let fieldNames: string[] = Object.values(FileFieldNames);
-  if (numberOfUploadedFiles > 0) {
-    for (let index = 0; index < numberOfUploadedFiles; index++) {
+  fileSet.forEach((file, index) => {
+    const baseFieldName = `${FileFieldNames.FILE_SET}[${index}]`;
+    if (file.administrativeAgreement) {
+      fieldNames.push(`${baseFieldName}.${SpecificFileFieldNames.ADMINISTRATIVE_AGREEMENT}`);
+    } else {
       for (const fileField of Object.values(SpecificFileFieldNames)) {
-        fieldNames.push(`${FileFieldNames.FILE_SET}[${index}].${fileField}`);
+        fieldNames.push(`${baseFieldName}.${fileField}`);
       }
     }
-  }
+  });
+  return fieldNames;
+};
+
+export const getAllContributorFields = (contributors: Contributor[]) => {
+  let fieldNames: string[] = Object.values(ContributorFieldNames);
+  contributors.forEach((contributor, index) => {
+    const baseFieldName = `${ContributorFieldNames.CONTRIBUTORS}[${index}]`;
+    for (const fileField of Object.values(SpecificContributorFieldNames)) {
+      // Don't include email field if user is not corresponding
+      if (!SpecificContributorFieldNames.EMAIL || contributor.corresponding) {
```

Comment:
skjønner ikke helt betydningen av kommentaren her. Hvis du ikke har email eller contributor.corresponding = true, så pusher du til fieldNames? Skal det være `||` eller `&&` ?

Hva sjekker du egentlig i if'en?

Reply:
haha, slet med å få dette til å fremstå logisk ja...
`SpecificContributorFieldNames.EMAIL` skal utelates hvis contributor ikke er `corresponding` (siden alternativet da er skjult).
Kanskje enklere med `if !(SpecificContributorFieldNames.EMAIL && !contributor.corresponding)`?

kune evt brukt if-else også, men da må vi liste opp alle felter manuelt igjen, så det blir en del ekstra linjer

Reply:
nei, waaat
dette er jo helt på bærtur
`SpecificContributorFieldNames.EMAIL` er jo alltid `true`

Reply:
😅

Reply:
fjerna hele `for (const fileField of Object.values(SpecificContributorFieldNames)) { }`.
tror det blir mer lesbart da, du får dømme!